### PR TITLE
WIP: github: use the new $GITHUB_ENV file instead of ::set-env

### PR DESCRIPTION
**Pull request by bazsi**

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -17,24 +17,23 @@ jobs:
...
             --disable-smtp
             --enable-all-modules
           "
-
-          echo "::set-env name=PYTHONUSERBASE::`echo ${PYTHONUSERBASE}`"
-          echo "::set-env name=PKG_CONFIG_PATH::`echo ${PKG_CONFIG_PATH}`"
-          echo "::set-env name=PATH::`echo ${PATH}`"
-          echo "::set-env name=THREADS::`echo ${THREADS}`"
-          echo "::set-env name=CONFIGURE_FLAGS::`echo ${CONFIGURE_FLAGS}`"
+          echo "/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:${PYTHONUSERBASE}/bin" >> $GITHUB_PATH
```
Comment (alltilla) [inline .github/workflows/macos.yml:36]: `PYTHONUSERBASE` is empty here, right?

File: .github/workflows/devshell.yml [hunk 2]
```diff
@@ -18,27 +18,27 @@ jobs:
...
-          export PYTHONVERSION=`printf ${{ matrix.python }} | tail -c 1`
-          export PYTHONUSERBASE=${HOME}/python_packages
-          export PATH=${PYTHONUSERBASE}:${PATH}
-          export CC=${{ matrix.cc }}
-          export CONFIGURE_FLAGS="
+          gh_export() {
+            echo "$1<<__EOF__" >> $GITHUB_ENV
+            echo "$2"          >> $GITHUB_ENV
+            echo "__EOF__"     >> $GITHUB_ENV
+          }
```
Comment (alltilla) [inline .github/workflows/devshell.yml:25]: Can we add an `export $1=$2`, and use the named variables below? E.g.: `PYTHONVERSION`

File: .github/workflows/devshell.yml [hunk 3]
```diff
@@ -144,7 +157,7 @@ jobs:
         run: ./autogen.sh
 
       - name: configure
-        run: ./configure ${DISTCHECK_CONFIGURE_FLAGS}
+        run: ./configure --disable-all-modules
```
Comment (alltilla) [inline .github/workflows/devshell.yml:160]: This change is not needed, IMO.

File: .github/workflows/macos.yml (hunk 1 above)
Comment (alltilla) [reply .github/workflows/macos.yml:41]: Fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by bazsi** [comment]
I am going to close this in favour of #3487 and reopen it with the bits that start with --disable-all-modules for the initial configure run of distcheck.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by gaborznagy** [comment]
@bazsi  I'll drop the patch from my PR #3476 setting `--disable-all-modules` in `make dist` since you are in progress fixing that (my patch: https://github.com/syslog-ng/syslog-ng/pull/3476/commits/fab60a03e8978eb8004d97fd38f7435a18774ad8).
Currently my PR depends on fixes from your PR (e.g. the manpage fix patch).

I'll just focus on a  working `distcheck` job in this PR(meaning I still have to disable java as agreed, and maybe other modules as well).

Alternatively, we can merge the two branches.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
closing as #3493 supersedes this.

**Pull request closed without merging**